standalone/test/integration_tests/api/integrationtest_scanner_api.cpp

```diff
@@ -149,30 +155,94 @@ void ScannerAPITests::prepareScannerMockStartReply()
+  }
   return start_req_received_barrier;
 }
 
+std::unique_ptr<util::Barrier> ScannerAPITests::prepareMonitoringFrameBarrier(
+    const std::vector<data_conversion_layer::monitoring_frame::Message>& msgs)
+{
+  const auto timestamp{ util::getCurrentTime() };
+  const auto scan{ createReferenceScan(msgs, timestamp) };
+
+  auto monitoring_frame_barrier = std::make_unique<util::Barrier>();
+
+  EXPECT_CALL(user_callbacks_,
+              LaserScanCallback(AllOf(ScanDataEqual(scan), TimestampInExpectedTimeframe(scan, timestamp))))
+      .WillOnce(OpenBarrier(monitoring_frame_barrier.get()));
+  return monitoring_frame_barrier;
+}
+
+void ScannerAPITests::startScanner()
+{
+  prepareScannerMockStartReply();
+  if (strict_scanner_mock_)
+  {
+    strict_scanner_mock_->startListeningForControlMsg();
+  }
+  else
+  {
+    nice_scanner_mock_->startListeningForControlMsg();
+  }
+  const auto start_future = scanner_->start();
+  expectIsDone("Scanner::start()", start_future);
+}
+
+void ScannerAPITests::expectNotDoneYet(const std::string& method_name, const std::future<void>& future)
+{
+  EXPECT_EQ(future.wait_for(FUTURE_WAIT_TIMEOUT), std::future_status::timeout) << method_name << " finished to early";
+}
+
+void ScannerAPITests::expectIsDone(const std::string& method_name, const std::future<void>& future)
+{
+  EXPECT_EQ(future.wait_for(DEFAULT_TIMEOUT), std::future_status::ready) << method_name << " not finished";
+}
+
+void ScannerAPITests::sendMonitoringFrames(const std::vector<data_conversion_layer::monitoring_frame::Message>& msgs)
+{
+  for (const auto& msg : msgs)
+  {
+    if (strict_scanner_mock_)
+    {
+      strict_scanner_mock_->sendMonitoringFrame(msg);
+    }
+    else
+    {
+      nice_scanner_mock_->sendMonitoringFrame(msg);
+    }
+    // Sleep to ensure that message are not sent too fast which might cause messages overwrite in socket buffer
+    std::this_thread::sleep_for(10ms);
+  }
+}
+
 TEST_F(ScannerAPITests, testStartFunctionality)
 {
   setUpScannerConfig();
   setUpScannerV2();
   setUpStrictScannerMock();
-  const auto start_req_received_barrier = prepareStrictMockStartRequestBarrier(*config_);
+  const auto start_req_received_barrier = prepareStartRequestBarrier(*config_);
 
   strict_scanner_mock_->startListeningForControlMsg();
   const auto start_future = scanner_->start();
 
   ASSERT_TRUE(start_req_received_barrier->waitTillRelease(DEFAULT_TIMEOUT)) << "Start request not received";
-  ASSERT_EQ(start_future.wait_for(FUTURE_WAIT_TIMEOUT), std::future_status::timeout) << "Scanner::start() finished too "
-                                                                                        "early";
+  expectNotDoneYet("Scanner::start()", start_future);
   strict_scanner_mock_->sendStartReply();
-  ASSERT_EQ(start_future.wait_for(DEFAULT_TIMEOUT), std::future_status::ready) << "Scanner::start() not finished";
+  expectIsDone("Scanner::start()", start_future);
```

Comment:
Using a future allready enables concurrent usage. Why not just wait for `start_future` and remove the barrier?